packages/apps/dashboard/server/src/modules/details/details.service.ts

```diff
@@ -148,40 +153,88 @@ export class DetailsService {
     return result;
   }
 
-  public async getLeadersByChainId(
+  public async getLeaders(
```

Comment:
Maybe for convenience we make pagination options as object?

```
type GetLeadersPaginationOptions = {
  orderBy?: LeadersOrderBy;
  orderDirection?: OrderDirection;
  first?: number;
  skip?: number,
}
...
public async getLeaders(chainId: ChainId, {
  orderBy,
  orderDirection,
  first,
  skip
}: GetLeadersPaginationOptions = {})
```
It's fine to have `chainId` as required argument because there is `ChainId.ALL` option we can explicitly use

Reply:
@dnechay I agree to include a pagination object)

Could you clarify what you mean by the ChainId.ALL option? Using ChainId.ALL at the SDK level results in an error. I only see this option:
```
const chainIds = chainId === ChainId.ALL
    ? MAINNET_CHAIN_IDS
    : chainId
    ? [chainId]
    : await this.networkConfig.getAvailableNetworks();
```

Reply:
@eugenvoronov 
<img width="225" alt="Screenshot 2024-12-16 at 15 05 42" src="https://github.com/user-attachments/assets/6678d452-77d2-4e0b-b068-62b1b9c68ad9" />

But from [this discussion](https://github.com/humanprotocol/human-protocol/pull/2904#discussion_r1881926672) my understanding is that no need to allow it, only specific chain to make leaders retrieval simpler